internal/api/cluster.go

```diff
@@ -59,33 +100,93 @@ func handleGetClusters(c *Context, w http.ResponseWriter, r *http.Request) {
 	outputJSON(c, w, clusters)
 }
 
-// handleCreateCluster responds to POST /api/clusters
+// handleCreateCluster responds to POST /api/clusters, beginning the process of creating a new
+// cluster.
 func handleCreateCluster(c *Context, w http.ResponseWriter, r *http.Request) {
 	createClusterRequest, err := newCreateClusterRequestFromReader(r.Body)
 	if err != nil {
-		c.Logger.WithField("error", err).Error("failed to decode request")
+		c.Logger.WithError(err).Error("failed to decode request")
 		w.WriteHeader(http.StatusBadRequest)
 		return
 	}
 
-	cluster, err := c.Provisioner.CreateCluster(createClusterRequest)
+	cluster := model.Cluster{
+		Provider:    createClusterRequest.Provider,
+		Provisioner: "kops",
+		Size:        createClusterRequest.Size,
+		State:       model.ClusterStateCreationRequested,
+	}
+	cluster.SetProviderMetadata(model.AWSMetadata{
```

Comment:
I'm not fully sold on this `*Metadata` modelling business. For now, I use this to pass `Zones` downstream, but it's a bit all over the place. Thinking we might want to converge on just a single `Metadata` field at least.